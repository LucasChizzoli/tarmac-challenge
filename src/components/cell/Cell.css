.cell-wrapper {
  flex-basis: auto;
  width: calc(100%/8);
  height: 12vw;
}

.cell-image {
  width: 100%;
  height: 100%;
}