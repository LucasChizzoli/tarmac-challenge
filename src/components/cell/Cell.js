import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { PICTURE_URL } from '../../endpoints';
import imageNotFound from '../../imageNotFound.jpg';
import './Cell.css';

const Cell = (props) => {
  const { member, id } = props;
  return (
    <div className="cell-wrapper">
      <Link className="nav-item" to={`member/${id}`}>
        <div className="cell-over-name">{member.name}</div>
        <img alt={member.pic} src={`${PICTURE_URL}${member.pic}.png`} className="cell-image" onError={(e) => { e.target.src = imageNotFound }} />
      </Link>
    </div>
  )
};

Cell.propTypes = {
  member: PropTypes.object.isRequired,
  id: PropTypes.number.isRequired
};

export default Cell;