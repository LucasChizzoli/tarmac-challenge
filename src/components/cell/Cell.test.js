import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../EnzymeSetup.js';

import Cell from './Cell';

describe('<Cell />', () => {

  const member = {};

  describe('render()', () => {
    it('renders the component', () => {
      const wrapper = shallow(<Cell member={member} id={0}/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('onClick()', () => {
    it('Successfully call onClick event', () => {
      const mockOnClick = jest.fn();
      const wrapper = shallow(<Cell onClick={mockOnClick} member={member} id={0}/>);

      wrapper.find('Link').simulate('click', { button: 0 });

      console.log(wrapper.find('Link'));

      expect(mockOnClick.mock.calls.length).toEqual(1);
    });
  });
});