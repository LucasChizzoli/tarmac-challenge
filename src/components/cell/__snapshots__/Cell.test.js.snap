// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> render() renders the component 1`] = `
<div
  className="cell-wrapper"
>
  <Link
    className="nav-item"
    replace={false}
    to="member/0"
  >
    <img
      className="cell-image"
      onError={[Function]}
      src="https://tarmac.io/assets/members/undefined.png"
    />
  </Link>
</div>
`;
